core/src/main/scala/io/finch/request/package.scala

```diff
@@ -162,6 +180,8 @@ package object request {
       def apply[Req](req: Req)(implicit ev: Req => HttpRequest) =
         try number.toFuture
         catch { case _: IllegalArgumentException => ValidationFailed(param, rule).toFutureException }
+
+      override def toString: String = s"param: $param"
```

Comment:
I'm not sure about `("a", "is positive")`. It was my plan to think about reuse of validators which would be neat to have in larger projects (I can do this in a separate PR so that we can get this through quicker). In that case you cannot hardcode any parameter names into the validator. I think I would prefer to have a single-item reader add the name of the item automatically while the combined ones do not. This is because if you write a validator for more than one parameter (which is the less common case anyway) you would probably want to write out the names yourself in the message (e.g. startDate should be smaller than endDate). This is hardly something the lib can figure out on its own. Does this make sense?